antarest/study/storage/variantstudy/model/command/remove_renewables_cluster.py

```diff
@@ -29,116 +30,95 @@ def __init__(self, **data: Any) -> None:
+
+        remove_area_cluster_from_binding_constraints(
+            study_data, self.area_id, self.cluster_id
+        )
 
         return (
             CommandOutput(
                 status=True,
-                message=f"Renewables cluster '{self.cluster_id}' removed from area '{self.area_id}'",
+                message=(
+                    f"Renewable cluster '{self.cluster_id}' removed"
+                    f" from the area '{self.area_id}'."
+                ),
             ),
-            dict(),
+            {},
         )
 
     def _apply(self, study_data: FileStudy) -> CommandOutput:
-        if self.area_id not in study_data.config.areas:
-            return CommandOutput(
-                status=False,
-                message=f"Area '{self.area_id}' does not exist",
-            )
-
-        if (
-            len(
-                [
-                    cluster
-                    for cluster in study_data.config.areas[
-                        self.area_id
-                    ].renewables
-                    if cluster.id == self.cluster_id
-                ]
-            )
-            == 0
-        ):
-            return CommandOutput(
-                status=False,
-                message=f"Renewables cluster '{self.cluster_id}' does not exist",
-            )
-
-        if len(study_data.config.areas[self.area_id].renewables) == 1:
-            study_data.tree.delete(
-                [
-                    "input",
-                    "renewables",
-                ]
-            )
-        else:
-            study_data.tree.delete(
-                [
-                    "input",
-                    "renewables",
-                    "clusters",
-                    self.area_id,
-                    "list",
-                    self.cluster_id,
-                ]
-            )
-            study_data.tree.delete(
-                [
-                    "input",
-                    "renewables",
-                    "series",
-                    self.area_id,
-                    self.cluster_id,
-                ]
-            )
-
-        self._remove_renewables_cluster(study_data.config)
-
-        return CommandOutput(
-            status=True,
-            message=f"Renewables cluster '{self.cluster_id}' removed from area '{self.area_id}'",
-        )
+        """
+        Applies the study data to update renewable cluster configurations and saves the changes:
+        remove corresponding the configuration and remove the attached time series.
+
+        Args:
+            study_data: The study data to be applied.
+
+        Returns:
+            The output of the command execution.
+        """
+        # It is required to delete the files and folders that correspond to the renewable cluster
+        # BEFORE updating the configuration, as we need the configuration to do so.
+        # Specifically, deleting the time series uses the list of renewable clusters from the configuration.
+        # fmt: off
```

Comment:
we should consider in a future Issue configuring Black to handle such cases where auto-formatting is ugly